File: ruby/strategy_pattern/rubber_duck.rb
# frozen_string_literal: true

require_relative 'duck'
# A fun duck shaped bathtub toy
class com.randylayne.designpatterns.strategy.RubberDuck < com.randylayne.designpatterns.strategy.Duck
  def initialize
    super(com.randylayne.designpatterns.strategy.QuackBehavior::com.randylayne.designpatterns.strategy.Squeak, com.randylayne.designpatterns.strategy.FlyBehavior::NoFly)
  end

  def display
    puts "I'm a rubber ducky"
  end
end

File: java/com/randylayne/designpatterns/strategy/RubberDuck.java
package com.randylayne.designpatterns.strategy;

public class RubberDuck extends Duck {
   public RubberDuck() {
      quackBehavior = new Squeak();
      flyBehavior = new FlyNoWay();
   }

   @Override
   public void display() {
      System.out.println("I'm a rubber ducky");
   }
}

